node/project-management-app/controllers/app.js

```diff
@@ -0,0 +1,288 @@
+        return this.handleHelp();
+      default:
+        return {};
+    }
+  }
+
+  /**
+   * Handles the ADDED_TO_SPACE event by sending back a welcome text message.
+   * It also adds the space to storage so it can later receive user stories.
+   * @return {Object} A welcome text message to post back to the DM or space.
+   */
+  async handleAddedToSpace() {
+    await SpaceService.createSpace(
+      this.spaceName, this.event.space.displayName);
+    const message = 'Thank you for adding the Project Manager app.' +
+      ' Message the app for a list of available commands.';
+    return { text: message };
+  }
+
+  /**
+   * Handles the REMOVED_FROM_SPACE event by deleting the space from storage.
+   */
+  async handleRemovedFromSpace() {
+    await SpaceService.deleteSpace(this.spaceName);
+    return {};
+  }
+
+  /**
+   * Handles a slash command and returns a message as a response.
+   * @return {Promise<Object>} A message to post back to the DM or space.
+   */
+  async handleSlashCommand() {
+    switch (Number(this.event.message.slashCommand.commandId)) {
+      case SlashCommand.CREATE_USER_STORY:
+        return this.handleCreateUserStory();
+      case SlashCommand.MY_USER_STORIES:
+        return this.handleMyUserStories();
+      case SlashCommand.USER_STORY:
+        return this.handleUserStory();
+      case SlashCommand.MANAGE_USER_STORIES:
+        return this.handleManageUserStories();
+      case SlashCommand.CLEANUP_USER_STORIES:
+        return this.handleCleanupUserStories();
+      default:
+        return { text: '⚠️ Unrecognized command.' };
+    }
+  }
+
+  /**
+   * Handles the create user story command.
+   * @return {Promise<Object>} A message to post back to the DM or space.
+   */
+  async handleCreateUserStory() {
+    const title = (this.event.message.argumentText || '').trim();
+    if (title.length === 0) {
+      return {
+        text: 'Title is required.'
+          + ' Include a title in the command: */createUserStory* _title_'
+      };
+    }
+    const description = await AIPService.generateDescription(title);
+    const userStory =
+      await UserStoryService.createUserStory(
+        this.spaceName, title, description);
+    return {
+      text: `<${this.userName}> created a user story.`,
+      cardsV2: [{
+        cardId: 'userStoryCard',
+        card: new UserStoryCard(userStory)
+      }]
+    };
+  }
+
+  /**
+   * Handles the my user stories command.
+   * @return {Promise<Object>} A message to post back to the DM or space.
+   */
+  async handleMyUserStories() {
+    const userStories =
+      await UserStoryService.listUserStoriesByUser(
+        this.spaceName, this.userName);
+    const openUserStories = userStories
+      .filter((userStory) => userStory.data.status !== Status.COMPLETED);
+    // Obtain a unique list of users assigned to the fetched user stories.
+    const userIds = [...new Set(openUserStories
+      .filter((userStory) => !!userStory.data.assignee)
```

Comment:
You're right, and looking at it further, ww don't even need to query the user from the database in this case, because all the listed stories here are assigned to the user who sent the message, and their display name and avatar is already provided in the Chat event. So, we can just populate the card with the user data from the event and avoid an extra database query. Fixed.